src/zlib_h.jl

```diff
@@ -1,11 +1,14 @@
 # general zlib constants, definitions
 
-@unix_only    const _zlib = "libz"
-@windows_only const _zlib = "zlib1"
+@static if is_unix()
```

Comment:
@vtjnash, the deprecation message suggests to use `@static` (`WARNING: `@windows_only` is deprecated, use @static if is_windows() instead`). Maybe we should update that as well. 

@tkelman, could you please clarify what is underdocumented?